### PR TITLE
MULE-18328: ComponentAst#getParameters must return parameters in the …

### PR DESCRIPTION
**Pull request by mlischetti**
…same order

### CODE REVIEW

File: modules/spring-config/src/test/java/org/mule/runtime/config/model/ComponentModelTestCase.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+@Story(PARAMETER_AST)
+public class ComponentModelTestCase extends AbstractMuleTestCase {
+
+  private static final String PARAMETER_A = "a";
+  private static final String PARAMETER_B = "b";
+  private static final String PARAMETER_C = "c";
+  private ClassTypeLoader TYPE_LOADER = ExtensionsTypeLoaderFactory.getDefault().createTypeLoader();
+
+  @Before
+  public void before() {}
```
Comment (elrodro83) [inline modules/spring-config/src/test/java/org/mule/runtime/config/model/ComponentModelTestCase.java:47]: remove

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java [hunk 2]
```diff
@@ -83,7 +84,7 @@
   private ComponentIdentifier identifier;
   private String componentId;
   private final Map<String, String> parameters = new HashMap<>();
-  private final Map<String, ComponentParameterAst> parameterAsts = new HashMap<>();
+  private Map<String, ComponentParameterAst> parameterAsts = new HashMap<>();
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java:87]: initialize this as `Collections.emptyMap()`

File: modules/spring-config/src/test/java/org/mule/runtime/config/model/ComponentModelTestCase.java (hunk 1 above)
Comment (mlischetti) [reply modules/spring-config/src/test/java/org/mule/runtime/config/model/ComponentModelTestCase.java]: done

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java (hunk 2 above)
Comment (mlischetti) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java]: replaced by map + list

File: modules/spring-config/src/test/java/org/mule/runtime/config/model/ComponentModelTestCase.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import org.mule.tck.junit4.AbstractMuleTestCase;
+import org.yaml.snakeyaml.parser.ParserException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Feature(ARTIFACT_AST)
+@Story(PARAMETER_AST)
+public class ComponentModelTestCase extends AbstractMuleTestCase {
```
Comment (elrodro83) [inline modules/spring-config/src/test/java/org/mule/runtime/config/model/ComponentModelTestCase.java:46]: rename to `ComponentAstParametersTestCase`

**Pull request merged**